src/test/test_proto_client.c

```diff
@@ -115,12 +117,22 @@ test_run()
 	rc = crt_group_rank(NULL, &test.tg_my_rank);
 	D_ASSERTF(rc == 0, "crt_group_rank() failed. rc: %d\n", rc);
 
-	/* Attempt to register actual fmt_0 and fmt_1 */
-	rc = crt_proto_register(&my_proto_fmt_0);
-	D_ASSERTF(rc == 0, "registration failed with rc: %d\n", rc);
-
-	rc = crt_proto_register(&my_proto_fmt_1);
-	D_ASSERTF(rc == 0, "registration failed with rc: %d\n", rc);
+	switch (test.tg_num_proto) {
+		case 4:
+			rc = crt_proto_register(&my_proto_fmt_3);
+			D_ASSERT(rc == 0);
+		case 3:
+			rc = crt_proto_register(&my_proto_fmt_2);
+			D_ASSERT(rc == 0);
+		case 2:
+			rc = crt_proto_register(&my_proto_fmt_1);
+			D_ASSERT(rc == 0);
+		case 1:
+			rc = crt_proto_register(&my_proto_fmt_0);
+			D_ASSERT(rc == 0);
+		default:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment